src/frontend/services/users-service.js

```diff
@@ -94,12 +98,21 @@ export class UsersService {
       `/api/users?usernames=${queryString}`
     );
 
-    // #region Mock data
-    // TODO: remove this region when data comes from BE
-    users.data.forEach((user) => {
-      user.avatarLink = "";
+    const cookie = CookieService.getCookie();
+
+    let basicAuth = window.btoa(cookie.username + ":" + cookie.password);
+
+    users.data = users.data.map((user) => {
+      if (user.avatarLink === null) {
+        return user;
+      }
+      return {
+        ...user,
+        avatarLink: `${
+          user.avatarLink
+        }?access_token=${basicAuth.toString()}&cacheInvalidator=${new Date().toISOString()}`, // Note that cache invalidator is needed because even if the image changes the link will remain the same. Adding that part makes the browser refetch the image.
```

Comment:
access_token va tolto in questo caso

Reply:
Visto che il link viene utilizzato all'interno degli elementi img serviva per autenticare l'utente, altrimenti chiunque potrebbe vedere l'immagine associata ad un determinato id.
Se non è un problema metto l'endpoint di quell'api come senza autenticazione.

Oppure posso fare la chiamata da fuori l'elemento, per farla autenticata normalmente.

Per favore dimmi quale delle 2 è meglio.

Reply:
hai ragione... lascia così